Feature/modal register

wandersoncbjr opened the pull request:
**usuário logado:**

![image](https://github.com/wandersoncbjr/e-commerce-taqtile/assets/101597114/b7ede65a-f867-48c1-a63a-1f87cb80a0fa)
____________________________________________________________________
**usuário deslogado:**

![image](https://github.com/wandersoncbjr/e-commerce-taqtile/assets/101597114/fa85da4e-0858-424d-b699-b01df5d14948)
____________________________________________________________________
**A seta de voltar é exibida quando o usuário acessa a tela de login:**

https://github.com/wandersoncbjr/e-commerce-taqtile/assets/101597114/93495f27-5b0f-45ab-af2b-e4075c303773

wandersoncbjr committed:
fix: icon menu mobile

wandersoncbjr committed:
feat: add icon close

wandersoncbjr committed:
feat: add component Modal register

wandersoncbjr committed:
fix: ContainerLink

wandersoncbjr committed:
fix: props button

wandersoncbjr committed:
fix: props modal

wandersoncbjr committed:
feat: modal login register

caslujpg commented on src/assets/icons/icon-menu.tsx:
```diff
@@ -1,6 +1,11 @@
-export function IconMenu() {
+interface IconMenuProps {
+  onClick: () => void;
```
pra que passar o onClick pro svg se e o bottao que envolta ele que deve ser clicado??

wandersoncbjr replied:
não entendi, tenho que envolver o svg em um button?

caslujpg commented on src/components/link-button/styles.tsx:
```diff
@@ -9,6 +9,7 @@ export const ContainerLink = styled.button<LinkButtonProps>`
   display: flex;
   border: none;
   gap: 4px;
+  justify-content: center;
```
pq antes nao precisava e agora precisa, isso nao quebra os outros lugares que utilizam??

wandersoncbjr replied:
Não precisa antes passou batido por aí corrigi. verifiquei se tinha algo quebrado em outros lugares, não vi nada

caslujpg commented on src/components/modal-register/modal-register.tsx:
```diff
@@ -0,0 +1,69 @@
+        <ContainerRegisterStyled>
+          <WrapperRegisterStyled>
+            <IconClose onClick={() => setShowModalRegister(false)} />
+            {user ? (
```
e como vai saber qual e o user e quando ele existe ou nao?? porque isso tem que ser automatico ne

wandersoncbjr replied:
Não entendi, a respota não vem da api?
poderia me explicar melhor na call?

caslujpg commented on src/components/modal-register/modal-register.tsx:
```diff
@@ -0,0 +1,69 @@
+
+                  <LinkButton
+                    variant="default"
+                    expanded={location.pathname !== "/login" ? true : false}
```
na web nao muda rota ne, isso funciona?

wandersoncbjr replied:
funciona.
Depende da rota se for de login não.
se for de 'detalhes' sim

wandersoncbjr committed:
feat: create IconArrowLeft component with StyledIconButton

- Created the IconArrowLeft component using StyledIconButton for styling.
- Allows an optional click handler with MouseEventHandler.
- Enhanced component styling with StyledIconButton.

wandersoncbjr committed:
refactor: updated IconMenu to use MouseEventHandler

- Changed the onClick prop type from () => void to MouseEventHandler<HTMLButtonElement>.
- Updated IconMenu to use StyledIconButton for consistent styling.

wandersoncbjr committed:
refactor: updated Header component with conditional rendering

- Added conditional rendering for IconArrowLeft based on the route location.

wandersoncbjr committed:
refactor: updated ModalRegister component

- Removed location-related conditionals and simplified logic for rendering.

wandersoncbjr committed:
Merge pull request #19 from wandersoncbjr/feature/page-login

Feature/page login
